tests/RequestTest.php

```diff
@@ -34,15 +37,24 @@ public function testCreateRequest()
 		);
 
 		$request = new Request($apiAction);
-		$result = $request->createRequest('someToken', 'someSecret');
+		$result = $request->createRequest($token, $secret);
+
+		$fields = [
+			'timestamp' => $result['timestamp'],
+			'token' => $token,
+			'resourcetype' => $result['resourcetype'],
+			'actionid' => $result['actionid'],
+		];
+		$hmac = base64_encode(hash_hmac('sha256',implode('',$fields),$secret,true));
 
 		$this->assertGreaterThan(0, $result['timestamp']);
 		$this->assertEquals('someResourceId', $result['resourceid']);
 		$this->assertEquals('someIdentifier', $result['identifier']);
 		$this->assertEquals([], $result['parameters']);
 		$this->assertEquals('someActionId', $result['actionid']);
 		$this->assertEquals('someResourceType', $result['resourcetype']);
-		$this->assertGreaterThan(0, strlen($result['hmac']));
+		$this->assertGreaterThanOrEqual(2, $result['hmac_version']);
+		$this->assertEquals($hmac, $result['hmac']);
```

Comment:
I would prefer comparing against some fixed string here instead of re-implementing the HMAC in the test